pkg/services/hcloud/server/server.go

```diff
@@ -420,6 +645,32 @@ func (s *Service) reconcileLoadBalancerAttachment(ctx context.Context, server *h
 	return reconcile.Result{}, nil
 }
 
+func (s *Service) createServerFromImageNameOrURL(ctx context.Context) (*hcloud.Server, *hcloud.Image, error) {
+	if s.scope.HCloudMachine.Spec.ImageName != "" {
+		return s.createServerFromImageName(ctx)
+	}
+	return s.createServerFromImageURL(ctx)
+}
+
+func (s *Service) createServerFromImageURL(ctx context.Context) (*hcloud.Server, *hcloud.Image, error) {
+	// Validate that HCloudImageURLCommand is given
+	hm := s.scope.HCloudMachine
+	if s.scope.HCloudImageURLCommand == "" {
+		return nil, nil, fmt.Errorf("controller was started without --hcloud-image-url-command. Provisioning via ImageURL %q not possible", hm.Spec.ImageURL)
+	}
+
+	image, err := getServerImage(ctx, s.scope.HCloudClient, s.scope.HCloudMachine, s.scope.HCloudMachine.Spec.Type, preRescueOSImage)
+	if err != nil {
+		return nil, nil, fmt.Errorf("createServerFromImageURL: failed to get server image: %w", err)
```

Comment:
if this is permanent, it has to be a condition

Reply:
I create a condition and an event now:
```go 
	image, err := s.getServerImage(ctx, hm.Spec.ImageName)
	if err != nil {
		err = fmt.Errorf("failed to get server image: %w", err)
		msg := err.Error()
		record.Warn(hm, "FailedGetBootstrapData", msg)
		conditions.MarkFalse(hm, infrav1.ServerAvailableCondition,
			string(hm.Status.BootState), clusterv1.ConditionSeverityWarning,
			"%s", msg)
		return nil, nil, err
	}
```

Reply:
I think you still have to do that, yes!